modules/lang-painless/src/main/resources/org/elasticsearch/painless/org.elasticsearch.script.update.txt

```diff
@@ -18,3 +18,17 @@ class org.elasticsearch.painless.api.Json {
   String dump(def)
   String dump(def, boolean)
 }
+
+class org.elasticsearch.script.Metadata {
+    String getIndex()
+    String getId()
+    String getRouting()
+    long getVersion()
+    String getOp()
```

Comment:
We discussed having get and set op outside of the metadata on each script type that needs it, as it isn't really metadata about the document. What happened with that?

Reply:
If it's on the script, then it conflicts with any variables named `op`, so it would be a backwards compatibility issue.

After thinking about it some more, it seems like a high cost to introduce another concept in addition to `field` and `metadata` for this one-off.  Maybe there's a better way to handle it?